src/AutorestSwift/View Models/KeyValueViewModel.swift

```diff
@@ -185,26 +155,84 @@ struct KeyValueViewModel: Comparable {
         let type = signatureParameter.schema.type
         switch type {
         case .integer,
-             .boolean:
+             .boolean,
+             .number:
             return "String(\(value))"
         // For these types, a variable will be created in the method using the naming convention `{key|value}String`
         case .date,
              .unixTime,
              .dateTime,
              .byteArray:
             return "\(value)String"
-        case .number:
-            return signatureParameter.required ? "\(value)String" : "String(\(value))"
         case .choice,
              .sealedChoice:
             return "\(value).rawValue"
         case .array:
             return "\(value).map { String($0) }.joined(separator: \"\(signatureParameter.delimiter)\") "
+        case .duration:
+            return "DateComponentsFormatter().string(from: \(value)) ?? \"\""
         default:
             return "\(value)"
         }
     }
 
+    private static func getValue(for constantSchema: ConstantSchema, isSkipUrlEncoding: Bool) -> String {
+        let constantValue: String = constantSchema.value.value
+        let type = constantSchema.valueType.type
+
+        if type == .string,
+            isSkipUrlEncoding {
+            return "\"\(constantValue)\".removingPercentEncoding ?? \"\""
+        } else {
+            switch type {
+            case .date,
+                 .unixTime,
+                 .dateTime,
+                 .byteArray,
+                 .string:
+                return "\"\(constantValue)\""
+            case .number:
+                let numberSchema = constantSchema.valueType
+                let swiftType = numberSchema.swiftType()
+                if swiftType == "Decimal" {
+                    return "\(swiftType)(\(constantValue))"
+                } else {
+                    return "String(\(swiftType)(\(constantValue)))"
+                }
+            default:
+                return "String(\(constantValue))"
+            }
+        }
+    }
```

Comment:
Is there a reason this isn't just a function directly on `ConstantSchema`?

Reply:
Good suggestion. Have moved the code to ConstantSchema